src/merkle/smt/mod.rs

```diff
@@ -265,6 +285,60 @@ pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
         }
     }
 
+    /// Parallel implementation of [`SparseMerkleTree::compute_mutations()`].
+    ///
+    /// This method recursively tracks mutations across 8-depth subtrees from the bottom up,  
+    /// ultimately reconstructing the complete mutation set for the entire tree.
+    ///
+    /// The implementation is similar to [`SparseMerkleTree::build_subtrees_from_sorted_entries()`],
+    /// sharing the same constraint that the depth must be a multiple of 8.
+    #[cfg(feature = "concurrent")]
+    fn compute_mutations_subtree(
```

Comment:
```suggestion
    fn compute_mutations_concurrent(
```

Nit: Should this be "concurrent" to be in contrast to "compute_mutations_sequential"?